### ISSUE TITLE
libFLAME backend?

### ISSUE THREAD

**Issue by datnamer** [libdynd/libdynd#600]
http://www.cs.utexas.edu/~flame/web/libFLAME.html 

This looks interesting.

**Comment by insertinterestingnamehere** [comment]
Yes, I've already spent some time looking at a few different backends we could use to get good support for linear algebra operations with DyND arrays. I don't know if libflame/blis are what we'll end up using, but they are definitely one good option. We're not actively pursuing this at the moment, but it's a fantastic idea that I'd like to explore.

**Comment by datnamer** [comment]
Cool :)

**Comment by datnamer** [comment]
@insertinterestingnamehere @izaid  have you seen this https://github.com/wjakob/pybind11/?

**Comment by insertinterestingnamehere** [comment]
@datnamer yes, several people have mentioned it to us. It looks like an awesome project, though we'll still be sticking with Cython. Cython's cimport machinery allows us to seamlessly export low-level API's between Python modules without having to do anything fancy with capsules. As to my knowledge, there isn't a good way to do that with anything else. Cython also makes it a lot easier for Python users to move from DyND-oriented Python code to mixed Python/C++ constructs then to straight C++ code while maintaining a stable Python API for whatever they are working on.

**Comment by datnamer** [comment]
Gotcha.

"Easier for Python users to move from DyND-oriented Python code to mixed Python/C++ constructs then to straight C++ code"

Curious, whats the use case for this? Hacking on the library or writing user code?

**Comment by insertinterestingnamehere** [comment]
Definitely both. One of several things I'm working on right now is making it so that all of that works seamlessly within our own code. Making it work for users should be very easy from there. There's still a good bit of refactoring left before things work really well though.

**Comment by datnamer** [comment]
Cool. But I'd rather not learn cpp :) Is there any way to contribute/extend the library without writing cpp? Maybe AOT numba compilation + cpp interop?

**Comment by seibert** [comment]
Numba is unlikely to get the ability to fully interoperate with C++ in the near future.  There are a lot of C++ ABI issues that LLVM can't handle for us.  Maybe we can eventually use clang as a JIT (as the cling developers have) to deal with this.
